src/backend/joanie/badges/models.py

```diff
@@ -0,0 +1,166 @@
+        verbose_name=_("ID"),
+        help_text=_("Primary key for the badge as UUID"),
+        primary_key=True,
+        default=uuid.uuid4,
+        editable=False,
+    )
+
+    name = models.CharField(
+        verbose_name=_("Name"),
+        help_text=_("Should be filled with the badge name from configured provider"),
+        max_length=100,
+        editable=False,
+    )
+
+    description = models.TextField(
+        _("Description"),
+        help_text=_(
+            "Should be filled with the badge description from configured provider"
+        ),
+        editable=False,
+    )
+
+    provider = models.CharField(
+        _("Provider"),
+        help_text=_(
+            "Should be filled with configured provider name/code for this badge"
+        ),
+        max_length=3,
+        choices=get_badge_provider_choices().choices,
+        db_index=True,
+        editable=False,
+    )
+
+    iri = models.URLField(
+        _("IRI"),
+        help_text=_(
+            "Generated badge IRI, usually the URL pointing to the badge created for a provider"
+        ),
+        editable=False,
+    )
+
+    created_on = models.DateTimeField(
+        verbose_name=_("Created on"),
+        help_text=_("Date and time at which a document request was created"),
+        auto_now_add=True,
+        editable=False,
+    )
+
+    updated_on = models.DateTimeField(
+        verbose_name=_("Updated on"),
+        help_text=_("Date and time at which a document request was last updated"),
+        auto_now=True,
+        editable=False,
+    )
+
+    class Meta:
+        """Options for the Badge model"""
+
+        db_table = "joanie_badge"
+        ordering = ["name"]
+        unique_together = [
+            ["name", "provider"],
+        ]
+        verbose_name = _("Badge")
+        verbose_name_plural = _("Badges")
+
+    def __str__(self):
+        return f"{self.provider}: {self.name}"
+
+
+class IssuedBadge(models.Model):
+    """
+    Issued badges for a ressource.
+    """
+
+    id = models.UUIDField(
+        verbose_name=_("ID"),
+        help_text=_("Primary key for the badge as UUID"),
+        primary_key=True,
+        default=uuid.uuid4,
+        editable=False,
+    )
+
+    iri = models.URLField(
+        _("Issued badge IRI"),
+        help_text=_("Issued badge IRI, usually the URL pointing to issued badge"),
+        editable=False,
+    )
+
+    ressource_link = models.URLField(
+        _("Ressource link"),
+        help_text=_(
+            "Link to the ressource the badge has been issued for (e.g. a course session)"
+        ),
+        editable=False,
+    )
+
+    user = models.ForeignKey(
+        User,
+        verbose_name=_("User"),
+        related_name="issued_badges",
+        on_delete=models.CASCADE,
+        editable=False,
+    )
+
+    badge = models.ForeignKey(
+        Badge,
+        verbose_name=_("Badge"),
+        related_name="issued",
+        on_delete=models.CASCADE,
+        editable=False,
+    )
```

Comment:
I may be more important to copy the name and description of the badge at the time of issuing than to just link the badge here. The badge may change and the issued badge should be sure to keep its value at the time of issuance.

Reply:
Good point, thanks!

Reply:
We should check the provider policy on this, _i.e._ will the issued badge evolves with the badge definition or is it statically generated? :thinking:

Reply:
Or should we treat the issuer as an optional remote? Am I wrong if I say we could issue the badge ourselves with what we have here? The name and description on the badge in our database may change so that's enough to justify copying it on the IssuedBadge?

Reply:
Let's talk about this IRL!

Reply:
So it's the provider's responsibility to ensure issued badges immutability. For OBF this is the case.

Reply:
OBF do it for themselves but we should do it for ourselves? Conclusion, I still don't agree but I'm ok to let you do it the way you think it should be done.

Reply:
I've added the `assertion` field instead that store issued badge assertion proofs.